### PR TITLE
Add Rivets implementation

### PR DESCRIPTION
**Pull request by mikeric**
Squashed version of #1224.

- [x] Add it to the home page, under labs.
- [x] Make sure the code passes JSHint
- [x] Make sure the code passes JSCS.
- [x] Add all files needed to run the app to the repo (and gitignore). Right now it has some 404s.
- [x] Address all 22 failing tests.
- [x] Squash commits

### CODE REVIEW

**Comment by arthurvr** [comment]
Why creating a new PR for that? You can just do that in the existing one. Anyways, thanks!

**Comment by arthurvr** [comment]
I confirmed the tests are passing. Nice work @mikeric. For the reference, this would resolve #1180.

File: examples/rivets/index.html [hunk 1]
```diff
@@ -0,0 +1,125 @@
+<!doctype html>
+
```
Comment (arthurvr) [inline examples/rivets/index.html]: remove this line

File: index.html [hunk 2]
```diff
@@ -269,7 +269,9 @@
 								<li>
 									<a href="examples/react-backbone/" data-source="http://facebook.github.io/react/" data-content="This React example integrates Backbone for its model and router. It is a showcase of third-party library integration for developers wishing to use React together with a different JavaScript framework.">React + <br>Backbone.js</a>
 								</li>
-
+								<li>
```
Comment (arthurvr) [inline index.html]: I believe this should have a routing class?

File: examples/rivets/js/components/todos.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		this.addTodo = this.addTodo.bind(this);
+		this.todos = TodoStore;
+		this.newTodo = '';
+	};
+
+	// Adds a new todo item.
+	TodosController.prototype.addTodo = function (ev) {
+		ev.preventDefault();
+
+		if (this.newTodo && this.newTodo.trim()) {
```
Comment (passy) [inline examples/rivets/js/components/todos.js]: Can you make this explicit by referring to the length? Makes it easier to follow, in particular for those not intimately familiar with the JS truth table.

File: examples/rivets/index.html [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+			</button>
+		</script>
+
+		<section id="todoapp"></section>
+
+		<footer id="info">
+			<p>Double-click to edit a todo</p>
+
+			<p>
+				Written by <a href="http://github.com/mikeric">Michael Richards</a>
```
Comment (arthurvr) [inline examples/rivets/index.html]: Use https :)

File: examples/rivets/index.html [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+	<body>
+		<script type="text/template" id="todos">
+			<header id="header">
+				<h1>todos</h1>
+				<form rv-on-submit='addTodo'>
+					<input
+						id="new-todo"
+						placeholder="What needs to be done?"
+						rv-value='newTodo'
```
Comment (arthurvr) [inline examples/rivets/index.html]: Our HTML uses double quotes.

**Comment by mikeric** [comment]
@arthurvr Any other changes that need to be made here? :)

**Comment by sindresorhus** [comment]
When double-clicking a todo item the text should be highlighted so it can be edited right away, but if you try to so nothing happens.

Also needs to fix the merge conflict.

Otherwise lgtm.

**Comment by samccone** [comment]
ping ping @mikeric

File: examples/rivets/index.html [hunk 6]
```diff
@@ -0,0 +1,124 @@
+<!doctype html>
+<html lang="en" data-framework="rivets">
+	<head>
+		<meta charset="utf-8">
+		<title>Rivets • TodoMVC</title>
+		<link rel="stylesheet" href="node_modules/todomvc-common/base.css">
+		<link rel="stylesheet" href="node_modules/todomvc-app-css/index.css">
+	</head>
+
```
Comment (arthurvr) [inline examples/rivets/index.html]: remove this line

**Comment by arthurvr** [comment]
hey @mikeric still interested in taking this up?

**Comment by arthurvr** [comment]
ping @mikeric :)

**Comment by samccone** [comment]
It has been over 30 days, when someone wants to pick this up we will reopen

**Comment by altmind** [comment]
Guys, I made the changes and pushed them as new PR #1295 (as PR is from personal fork).
Hope, this finally be closed :dart:

**Pull request closed without merging**